### PR TITLE
Allow use of ocean obs prep in WCDA cycling and remove R2D2

### LINKED ISSUE #815: Remove R2D2 and replace with something similar to what is done for the atmosphere.

### PR DESCRIPTION
**Pull request by AndrewEichmann-NOAA**
# Description

Enables use of ocean obs prep task in WCDA cycling and removes R2D2 from same. Experiment setup is same as before, but runs task `gdasprepoceanobs` before `gdasocnanalprep`. This obtains ocean data nc4 files from `DMPDIR`, processes them into IODA format and copies them to `COM_OBS`. This replaces the current R2D2 processing.

This PR resolves https://github.com/NOAA-EMC/GDASApp/issues/815 and is mutual dependent on the recently merged to GDASApp https://github.com/NOAA-EMC/GDASApp/pull/836 

# Type of change

- New feature (adds functionality)

# Change characteristics
- Is this a breaking change (a change in existing functionality)? YES
- Does this change require a documentation update? NO

# How has this been tested?

WCDA cycling on Hera

# Checklist
- [x] Any dependent changes have been merged and published
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] New and existing tests pass with my changes
- [x] I have made corresponding changes to the documentation if necessary

### CODE REVIEW

**Comment by AndrewEichmann-NOAA** [comment]
FYI @guillaumevernieres and @CoryMartin-NOAA

File: parm/config/gfs/config.prepoceanobs [hunk 1]
```diff
@@ -7,11 +7,14 @@ echo "BEGIN: config.prepoceanobs"
...
 
 export OBS_YAML_DIR=${HOMEgfs}/sorc/gdas.cd/parm/soca/obs/config
-export OBSPROC_YAML=@OBSPROC_CONFIG@
+export OBSPROC_YAML=${HOMEgfs}/sorc/gdas.cd/parm/soca/obsproc/obsproc_config.yaml
 export OBS_LIST=@SOCA_OBS_LIST@
 [[ -n "${OBS_LIST}" ]] || export OBS_LIST=${HOMEgfs}/sorc/gdas.cd/parm/soca/obs/obs_list.yaml
 export OBS_YAML=${OBS_LIST}
 
+# until dmpdir has full ocean obs; works on Hera
+export DMPDIR=/scratch1/NCEPDEV/stmp4/Shastri.Paturi/forAndrew
```
Comment (guillaumevernieres) [inline parm/config/gfs/config.prepoceanobs]: Sorry, just thinking of this now. We could probably do something like 
```
export DMPDIR=@DMPDIR@
export OBSPROC_YAML=@OBSPROC_YAML@
...
```
and set these variables in the `yaml` we pass to that setup script, similarly to what we did with the R2D2 source.

```yaml
base:
 [stuff]
 ocnanal:
  [stuff]
 prepoceanobs:
   DMPDIR=/scratch1/NCEPDEV/stmp4/Shastri.Paturi/forAndrew
   OBSPROC_YAML=...
```

**Review by guillaumevernieres — COMMENTED** [review]
Thanks for doing this @AndrewEichmann-NOAA . See comment.

File: parm/config/gfs/config.prepoceanobs (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.prepoceanobs]: That's how `OBSPROC_YAML` was originally set, so shouldn't be an issue

File: workflow/rocoto/gfs_tasks.py [hunk 2]
```diff
@@ -588,6 +588,22 @@ def ocnanalprep(self):
         deps.append(rocoto.add_dependency(dep_dict))
         dependencies = rocoto.create_dependency(dep=deps)
 
+        resources = self.get_resource('prepoceanobs')
+        task = create_wf_task('prepoceanobs',
```
Comment (aerorahul) [inline workflow/rocoto/gfs_tasks.py]: This method is no longer available.

**Review by aerorahul — CHANGES_REQUESTED** [review]

File: workflow/rocoto/gfs_tasks.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/gfs_tasks.py]: Has its functionality been replaced with` rocoto.create_task`?

Comment (aerorahul) [reply workflow/rocoto/gfs_tasks.py]: yes.  for example, please take a look at the `ocnanalprep` task below that has been updated.

File: parm/config/gfs/config.prepoceanobs (hunk 1 above)
Comment (AndrewEichmann-NOAA) [reply parm/config/gfs/config.prepoceanobs]: done

**Review by guillaumevernieres — APPROVED** [review]
Looks good to me, thanks @AndrewEichmann-NOAA  .

File: workflow/rocoto/gfs_tasks.py (hunk 2 above)
Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/gfs_tasks.py]: Corrected, with a couple of bugfixes

Comment (AndrewEichmann-NOAA) [reply workflow/rocoto/gfs_tasks.py]: PS also tested with a full cycle of WCDA

**Review by aerorahul — APPROVED** [review]

**Pull request merged**